traits/has_traits.py

```diff
@@ -3577,6 +3582,13 @@ def wrapped_class(klass):
         if CHECK_INTERFACES:
             from .interface_checker import check_implements
 
+            warnings.warn(
+                (
+                    "In the future, the @provides decorator will no longer "
+                    "perform runtime interface checks."
```

Comment:
"runtime" is a bit misleading here, since these checks will typically happen at import time. (Which is technically a _part_ of runtime, but still misleading.)
```suggestion
                    "perform interface checks."
```

Reply:
Resolved.